src/utils/crypto-utils.ts

```diff
@@ -0,0 +1,20 @@
+import { createDecipheriv } from "crypto"
+
+import { config } from "@config/config"
+
+const ALGORITHM = "aes-256-cbc"
+
+export const decryptPassword = (encryptedPassword: string, iv: string) => {
+  const SECRET_KEY = Buffer.from(
+    config.get("aws.amplify.passwordSecretKey"),
+    "hex"
+  )
+  const decipher = createDecipheriv(
+    ALGORITHM,
+    SECRET_KEY,
+    Buffer.from(iv, "hex")
+  )
+  let decrypted = decipher.update(encryptedPassword, "hex", "utf8")
+  decrypted += decipher.final("utf8")
+  return decrypted
```

Comment:
sorry, but this is abit confusing to me - it seems to me that we are decrypting first and thereafter, converting from `hex` to `utf8` before calling a finalizer of `utf8` and appending it to the updated value (in effect, `decrypted === decipher<utf8> * 2)` isn't it?)  

am i misunderstanding something? shouldn't it be either 

```ts
// update -> final
decipher.update(...)
return decipher.final(...)

// just return straight
return decipher.final(...)
```

why have a repeated output string?

Reply:
This is the usage as given by the [documentation](https://nodejs.org/docs/latest-v18.x/api/crypto.html#class-decipher) - i'm not sure what to call the intermediate steps either tbh, so I used their naming. But .update alone doesn't actually provide the final decrypted string, and .final is needed to cut off the decipher usage, but doesn't actually provide any decryption